Sources/StructuredQueriesCore/Table.swift

```diff
@@ -3,9 +3,16 @@
 /// Don't conform to this protocol directly. Instead, use the `@Table` and `@Column` macros to
 /// generate a conformance.
 @dynamicMemberLookup
-public protocol Table: QueryRepresentable where TableColumns.QueryValue == Self {
+public protocol Table: QueryRepresentable, PartialSelectStatement {
+  associatedtype QueryValue = Self
+
+  associatedtype From = Never
+
   /// A type that describes this table's columns.
-  associatedtype TableColumns: TableDefinition
+  associatedtype TableColumns: TableDefinition<Self>
+
+  /// A type that describes this table as a query expression.
+  associatedtype Selection: TableExpression<Self>
```

Comment:
Bikeshed opportunity: this is aliased to `Columns`, but named `Selection` here to distinguish itself better from `TableColumns`. Do we want to consider alternate naming, though to avoid overloading things with aliases?